### ISSUE TITLE
Binary arithmetic with numpy arrays creates arrays of random variables

### ISSUE THREAD

**Issue by JonathanWenger** [probabilistic-numerics/probnum#10]
Due to the way element-wise way numpy implements binary arithmetic an operation such as
```
import numpy as np
from probnum.probability import RandomVariable, asrandomvariable, Normal
A = np.array([3, 2])
X = RandomVariable(distribution=Normal(mean=np.array([1, 2]), cov=np.array([[2, 0], [0, 5]])))
Y = A + X
```
produces an array of random variables instead of a multivariate normal random variable. This loses covariance information, while the following does not:
```
A = np.array([3, 2])
X = RandomVariable(distribution=Normal(mean=np.array([1, 2]), cov=np.array([[2, 0], [0, 5]])))
Y = asrandvar(A) + X
```
This is first and foremost an issue of operator precedence as the ``__radd__`` function of ``RandomVariable`` is not called, since numpy calls the ``__radd__`` function for each element individually. See also:
- https://stackoverflow.com/questions/22632937/how-to-make-numpy-overloading-of-add-independent-on-operand-order
- https://stackoverflow.com/questions/22289093/can-you-impose-object-precedence-for-overloaded-operators-in-python

One possible solution could be to prevent broadcasting when performing a binary operation with a dirac distribution. Raising a ``NotImplementedError`` may trigger the ``__radd__`` function correctly. This would mean that
```
Y = 1 + X
```
throws an error.

**Comment by JonathanWenger** [comment]
Proposed fix: ``__array_ufunc__ = None`` in ``RandomVariable``.

**Comment by JonathanWenger** [comment]
Fixed by 2ec7c278e3bb1760e116e16adda5ac6dc252a762.
